dbms/src/Functions/divide.cpp

```diff
@@ -62,7 +62,11 @@ struct TiDBDivideFloatingImpl<A, B, false>
     template <typename Result = ResultType>
     static Result apply(A a, B b)
     {
-        return static_cast<Result>(a) / b;
+        // round for decimal floating number dividing
+        if constexpr (std::is_integral_v<Result> || std::is_same_v<Result, Int256>)
+            return (static_cast<Result>(a) + b / 2) / b;
```

Comment:
does it still works when `a` or `b` is negative integer?

Reply:
I found a code snip which can handle negative case [https://stackoverflow.com/a/18067292](url)

Reply:
thanks, after test, this method can't support MAX and MIN, I pick another answer and changed a little.